code_generator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -9,493 +9,333 @@
+                entityManager.AddComponentData(entity, global::<#= qualifiedNamespace #>.<#= componentDetails.TypeName #>.Serialization.Deserialize(op.Data.SchemaData.Value.GetFields()));
 
-                var <#= componentDetails.CamelCaseTypeName #> = new <#= componentDetails.TypeName#>();
-<# foreach (var fieldDetails in fieldDetailsList) { #>
-                <#= componentDetails.CamelCaseTypeName#>.<#= fieldDetails.PascalCaseName #> = <#= UnityTypeMappings.GetNativeTypeMethod(fieldDetails.RawFieldDefinition, "data." + fieldDetails.CamelCaseName, enumSet) #>;
-<# } #>
-                <#= componentDetails.CamelCaseTypeName#>.DirtyBit = false;
-
-<# if (componentDetails.IsBlittable) { #>
-                view.AddComponent(entity, <#= componentDetails.CamelCaseTypeName#>);
-<# } else { #>
-                view.SetComponentObject(entity, <#= componentDetails.CamelCaseTypeName #>);
-<# } #>
-                view.AddComponent(entity, new NotAuthoritative<<#= componentDetails.TypeName #>>());
-
-                if (view.HasComponent<ComponentRemoved<<#= componentDetails.TypeName #>>>(entity))
+                if (entityManager.HasComponent<ComponentRemoved<<#= componentDetails.TypeName #>>>(entity))
                 {
-                    view.RemoveComponent<ComponentRemoved<<#= componentDetails.TypeName #>>>(entity);
+                    entityManager.RemoveComponent<ComponentRemoved<<#= componentDetails.TypeName #>>>(entity);
                 }
-                else if (!view.HasComponent<ComponentAdded<<#= componentDetails.TypeName #>>>(entity))
+                else if (!entityManager.HasComponent<ComponentAdded<<#= componentDetails.TypeName #>>>(entity))
                 {
-                    view.AddComponent(entity, new ComponentAdded<<#= componentDetails.TypeName #>>());
+                    entityManager.AddComponentData(entity, new ComponentAdded<<#= componentDetails.TypeName #>>());
                 }
-                else
+                else 
                 {
-                    LogDispatcher.HandleLog(LogType.Error, new LogEvent(
-                            "Received ComponentAdded but have already received one for this entity.")
+                    LogDispatcher.HandleLog(LogType.Error, new LogEvent(ReceivedDuplicateComponentAdded)
                         .WithField(LoggingUtils.LoggerName, LoggerName)
                         .WithField(LoggingUtils.EntityId, op.EntityId.Id)
-                        .WithField(MutableView.Component, "<#= componentDetails.TypeName #>"));
+                    );
                 }
             }
 
-            public void OnComponentUpdate(ComponentUpdateOp<<#= componentDetails.FullyQualifiedSpatialTypeName #>> op)
+            public override void OnRemoveComponent(RemoveComponentOp op)
             {
-                Unity.Entities.Entity entity;
-                if (!view.TryGetEntity(op.EntityId.Id, out entity))
+                if (MutableView.TryGetEntity(op.EntityId, out var entity))
```

Comment:
Can this ever happen? Some of the other errors a user could strictly cause. However you'd have to do pretty weird things to trigger this.